### ISSUE TITLE
xapian: sanity check that tier names to -z actually exist

### ISSUE THREAD

**Issue by brong** [cyrusimap/cyrus-imapd#2764]
Apparently xapian keeps on compressing if an invalid tier name is given as the target.  This should be rejected outright!

**Comment by dilyanpalauzov** [comment]
This ticket was created in the context of https://lists.andrew.cmu.edu/pipermail/cyrus-devel/2019-May/004479.html . I would not say the tier name was invalid, but there wasn’t a single file in the filesystem created for that tier.

The invocation was `squatter -v -z Y -t X -o` and afterwards twice the message `undefined search partition X,Ysearch-partition-default` appeared.  Then squatter continued with
```
- FIRST USER -
compressing X:0,X,Y:0 to Y:3 for … (active Y:0,X:0,X,Y:0,Y:1,Y:2)
only one source, copying directly to ...NEW
renaming tempdir into place
finished compact of … (active Y:0,Y:1,Y:2,Y:3)

- NEXT USER and all other users -
comperessing X:0 to Y:0 for … (active X:0)
adding new initial search location Y:1
only one source, copying directly to ...NEW
renaming tempdir into place
finished compact of user … (active Y:1,Y:0)
```
I have to say that before running this I tried different things, some of which I interrupted for the first user and therefore for it things a different.

Note, that for the second user, at the end the sequence is 1,0 and for the first user the sequence is 0,1,2,3.  Moreover for the first user Y:0 is mentioned twice as active.

All in all, partition Y has not existed before running the command and the purpose of the call was to rename X to Y and toprepend a new partition to the xapianactive file. This new partition is not mentioned here, but is the new default in the imapd.conf.

Using cyrus imap 3.0.9.

**Comment by brong** [comment]
No, that's not why this ticket was created.  It was created because you could run -z -X and it would create a xapianactive file with `temp:5 archive:0 -X:0`.

This particular issue was fixed badly by a259f8b38d12fcc8d04c7dc4d5f55d454854c2d5 and then fixed properly in 35ef88d83d27c42c8bb01c30156b8d043beabaae.

**Comment by brong** [comment]
Regarding the other issue - it's possible for failures of index or repack to create spurious entries in xapianactive that don't have files under them - that doesn't break anything.  When you compact them, the entries get removed.
